### PR TITLE
feat: [ANDROAPP-6034] Input User and Input Password

### PR DESCRIPTION
**Pull request by xavimolloy**

### CODE REVIEW

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputPassword.kt [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+import androidx.compose.ui.text.TextRange
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.input.TextFieldValue
+import androidx.compose.ui.text.input.VisualTransformation
+import org.hisp.dhis.mobile.ui.designsystem.theme.Spacing
+
+@Composable
+fun InputPassword(
```
Comment (andresmr) [inline designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputPassword.kt:30]: What do you think about creating a test asserting data is filled and deleted properly?
And what about a screenshot testing to preserve the design?

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputUser.kt [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import androidx.compose.material3.Icon
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.TextFieldValue
+
+@Composable
+fun InputUser(
```
Comment (andresmr) [inline designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputUser.kt:15]: What do you think about creating a test asserting data is filled and deleted properly?
And what about a screenshot testing to preserve the design?

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputPassword.kt [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        legend = {
+            uiModel.legendData?.let {
+                Legend(uiModel.legendData, Modifier.testTag("INPUT_PASSWORD_LEGEND"))
+            }
+        },
+        inputStyle = uiModel.inputStyle,
+    )
+}
+
+data class InputPasswordModel(
```
Comment (andresmr) [inline designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputPassword.kt:126]: Aren't InputPasswordModel and InputUserModel pretty similar? do you think it make sense to think on a common UI model or at least a whay we can all of them in a common list, I'm thinking if we want to have a list of elements (final ui model just to render) the list needs to contains different clases (InputPasswordModel, inputUserModel, InputDateModel...) so we need to map in the UI, but if there is a common class (like the same ui model or sealed models) we can map the list in the presentation layer.
What do you think?

**Review by Balcan — APPROVED** [review]

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputPassword.kt (hunk 1 above)
Comment (xavimolloy) [reply designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputPassword.kt:30]: of course! good idea!

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputUser.kt (hunk 2 above)
Comment (xavimolloy) [reply designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputUser.kt:14]: of course! good idea!

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputPassword.kt (hunk 3 above)
Comment (xavimolloy) [reply designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputPassword.kt]: updated PR

**Review by andresmr — APPROVED** [review]

**Pull request merged**